File: solutions/1147-flip-columns-for-maximum-number-of-equal-rows/flip-columns-for-maximum-number-of-equal-rows.cpp
class Solution {
public:
    int maxEqualRowsAfterFlips(vector<vector<int>>& matrix) {
        unordered_map<string, int> patterns;
        
        for (const auto& row : matrix) {
            string pattern;
            int first = row[0];
            
            for (int num : row) {
                pattern += (num == first ? '1' : '0');
            }
            patterns[pattern]++;
        }
        
        int maxCount = 0;
        for (const auto& [pattern, count] : patterns) {
            maxCount = max(maxCount, count);
        }
        
        return maxCount;
    }
};

File: solutions/1147-flip-columns-for-maximum-number-of-equal-rows/flip-columns-for-maximum-number-of-equal-rows.swift
class Solution {
    func maxEqualRowsAfterFlips(_ matrix: [[Int]]) -> Int {
        var count: [String: Int] = [:]
        
        for row in matrix {
            var key = ""
            for (i, num) in row.enumerated() {
                key += String(row[0] == 1 ? 1 - num : num)
            }
            count[key, default: 0] += 1
        }
        
        return count.values.max() ?? 0
    }
}